metrics.go

```diff
@@ -143,11 +150,13 @@ func OpenWriteFifo(path string) error {
 
 // OpenReadFifo opens the fifo file for reading, returning the reader
 func OpenReadFifo(path string) (io.ReadCloser, error) {
-	file, err := os.OpenFile(path, os.O_RDONLY|syscall.O_NONBLOCK, os.ModeNamedPipe)
+	file, err := os.OpenFile(
```

Comment:
nit: whitespace/wrapping changes in their own commit would make this PR much more reviewable

Reply:
Hardly a problem in this case.